osbs/utils/yaml.py

```diff
@@ -50,12 +53,22 @@ def read_yaml(yaml_data, schema, package=None):
         logger.error('unable to extract JSON schema, cannot validate')
         raise
 
+    # Load schema into py object
     try:
         schema = json.load(schema)
     except ValueError:
         logger.error('unable to decode JSON schema, cannot validate')
         raise
-    data = yaml.safe_load(yaml_data)
```

Comment:
why did you changed order (schema check, data loading)?

Reply:
I reasoned that there's no point in messing with the schema at all if there's an issue with loading the data to be verified.